notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/listeners/SchemaNotificationEventListener.java

```diff
@@ -0,0 +1,79 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events.listeners;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import com.expediagroup.streamplatform.streamregistry.core.events.NotificationEvent;
+import com.expediagroup.streamplatform.streamregistry.core.events.NotificationEventHandler;
+import com.expediagroup.streamplatform.streamregistry.core.events.NotificationEventListener;
+import com.expediagroup.streamplatform.streamregistry.model.Schema;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class SchemaNotificationEventListener implements NotificationEventListener<Schema> {
+    private static final String IS_CREATING_A_SCHEMA = "" +
+            "event.entity instanceof T(com.expediagroup.streamplatform.streamregistry.model.Schema)" +
+            "and event.eventType == T(com.expediagroup.streamplatform.streamregistry.core.events.EventType).CREATE";
+
+    private static final String IS_UPDATING_A_SCHEMA = "" +
+            "event.entity instanceof T(com.expediagroup.streamplatform.streamregistry.model.Schema)" +
+            "and event.eventType == T(com.expediagroup.streamplatform.streamregistry.core.events.EventType).UPDATE";
+
+    private static final String IS_DELETING_A_SCHEMA = "" +
+            "event.entity instanceof T(com.expediagroup.streamplatform.streamregistry.model.Schema)" +
+            "and event.eventType == T(com.expediagroup.streamplatform.streamregistry.core.events.EventType).DELETE";
+
+
+    private final List<NotificationEventHandler<Schema>> notificationEventHandlers;
+
+    @Override
+    @EventListener(condition = IS_CREATING_A_SCHEMA)
+    public void onCreate(NotificationEvent<Schema> event) {
+        log.debug("On update schema event {}", event);
+        notificationEventHandlers.parallelStream().forEach(h -> this.handle(h::onCreate, event));
+    }
+
+    @Override
+    @EventListener(condition = IS_UPDATING_A_SCHEMA)
+    public void onUpdate(NotificationEvent<Schema> event) {
+        log.debug("On update schema event {}", event);
+        notificationEventHandlers.parallelStream().forEach(h -> this.handle(h::onUpdate, event));
+    }
+
+    @Override
+    @EventListener(condition = IS_DELETING_A_SCHEMA)
+    public void onDelete(NotificationEvent<Schema> event) {
+        log.debug("On delete schema event {}", event);
+        notificationEventHandlers.parallelStream().forEach(h -> this.handle(h::onDelete, event));
+    }
+
+    private void handle(Consumer<NotificationEvent<Schema>> handle, NotificationEvent<Schema> event) {
+        try {
+            handle.accept(event);
```

Comment:
Maybe we should add some metrics here to monitor events succeeded/failed? Or perhaps in the Kafka impl.

Reply:
KafkaTemplate has a pretty convenient callback to process both success and error in the producer process. What kind of metrics are you considering?

Reply:
SerDe error, send success/failure.

Reply:
we can do that in a follow up

Reply:
Absolutely